client/src/components/Product_card/index.js

```diff
@@ -0,0 +1,71 @@
+
+import React, { Component } from 'react'
+import './style.css'
+import axios from 'axios';
+import { Link } from 'react-router-dom'
+
+
+class ProductCard extends React.Component {
+  state = {
+    products: []
+
+  }
+  
+
+  
+  componentDidMount() {
+    if(this.props.match.params.id)
+    {
+    axios.get(`/shop/${this.props.match.params.id}`)
+    .then( (response) => {
+      const resultArray =response.data.map(e=>e)
+      this.setState({products:resultArray},()=>{
+      })
+      
+    })}else{
+      axios.get("/shop")
+      .then( (response) => {
+        const resultArray =response.data.map(e=>e)
+        this.setState({products:resultArray},()=>{
+        })
+        
+      })
+
+
+
+    }
+  }
+  render () {
+  const {products} = this.state
+  
+    return (
+      <>
+      {!products.length ? (
+          
+          <img
+            className="loading"
+            src="https://media1.tenor.com/images/556e9ff845b7dd0c62dcdbbb00babb4b/tenor.gif?itemid=5300368"
+            alt="loading"
```

Comment:
in this case, the page will still be `loading` even there's no data in the DB to display,
you need to make it show a message after checking the response (if it's empty) to inform the user that there are no data to display.